extensions/GitExtension/FileExplorerGitIntegration/Models/GitCommandRunner.cs

```diff
@@ -0,0 +1,229 @@
+
+    public void ReadInstallPath()
+    {
+        GitExeInstallPath = fileService.Read<string>(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName);
+    }
+
+    public bool IsGitExeInstallPathSet()
+    {
+        return !string.IsNullOrEmpty(GitExeInstallPath);
+    }
+
+    public GitCommandRunnerResultInfo ValidateGitExeInstallPath()
+    {
+        try
+        {
+            System.Diagnostics.Process.Start(GitExeInstallPath);
+            return new GitCommandRunnerResultInfo(ProviderOperationStatus.Success, null);
+        }
+        catch (Exception ex)
+        {
+            log.Error(ex, "Failed to start Git.exe at configured path");
+            return new GitCommandRunnerResultInfo(ProviderOperationStatus.Failure, "Failed to start Git.exe at configured path", string.Empty, ex, null);
+        }
+    }
+
+    public bool StoreGitExeInstallPath(string path)
+    {
+        log.Information("Setting Git Exe Install Path");
+        GitExeInstallPath = path;
+        fileService.Save(GitExecutableConfigOptions.GitExecutableConfigFolderPath, GitExecutableConfigOptions.GitExecutableConfigFileName, GitExeInstallPath);
+        log.Information("Git Exe Install Path stored successfully");
+        return true;
+    }
+
+    public bool DetectGit()
+    {
+        var gitExeFound = false;
+
+        if (!gitExeFound)
+        {
+            // Check if git.exe is present in PATH environment variable
+            try
+            {
+                System.Diagnostics.Process.Start("git.exe");
+                gitExeFound = true;
+                StoreGitExeInstallPath("git.exe");
+            }
+            catch (Exception ex)
+            {
+                log.Debug(ex, "Failed to start Git.exe");
+            }
+        }
+
+        if (!gitExeFound)
+        {
+            // Check execution of git.exe by finding install location in registry keys
+            string[] registryPaths = { "HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Git_is1", "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Git_is1" };
+
+            foreach (var registryPath in registryPaths)
+            {
+                var gitPath = Registry.GetValue(registryPath, "InstallLocation", defaultValue: string.Empty) as string;
+                if (!string.IsNullOrEmpty(gitPath))
+                {
+                    var paths = FindSubdirectories(gitPath);
+                    CheckForExeInPaths(paths, ref gitExeFound);
+                    if (gitExeFound)
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (!gitExeFound)
+        {
+            // Search for git.exe in common file paths
+            var programFiles = Environment.GetEnvironmentVariable("ProgramFiles");
+            var programFilesX86 = Environment.GetEnvironmentVariable("ProgramFiles(x86)");
```

Comment:
>"ProgramFiles(x86) [](http://example.com/codeflow?start=69&length=18)

what about ProgramFiles(Arm)?

Reply:
Isn't this folder only for 32-bit Arm applications which future versions of Windows will not support? In this case, it looks like an additional check inside this dir is not required. Thanks!

Reply:
Dev Home runs on Windows 10 as well, which still supports 32-bit ARM I believe

Reply:
I can't find any packages that install 32-bit ARM Git for Windows. Is that a thing? Or are you thinking somebody running ARM32 DevHome is going to build git from source and install it?

Reply:
You're right there never was a 32-bit ARM package for git for windows, I had assumed there must be but it looks like arm support for git for windows is relatively new.